go/engine/paperkey_gen.go

```diff
@@ -264,5 +280,31 @@ func (e *PaperKeyGen) push(ctx *Context) error {
 		Contextified:   libkb.NewContextified(e.G()),
 	}
 
-	return libkb.DelegatorAggregator(ctx.LoginContext, []libkb.Delegator{sigDel, sigEnc}, nil)
+	var sdhBoxes = []libkb.SharedDHSecretKeyBox{}
+	if e.G().Env.GetEnableSharedDH() {
+		if e.getSharedDHKeyring().CurrentGeneration() == 0 {
```

Comment:
Introduce a helper function that checks for this legacy/upgraded case specifically, rather than checking `CurrentGeneration` for 0.